### PR TITLE
Adding the Dollar Sign Glyph to the token parser

### PR DESCRIPTION
**Pull request by dkhenry**
MySQL accepts Dollar Signs as part of string identifiers.

Signed-off-by: Dan Kozlowski <koz@planetscale.com>

### CODE REVIEW

File: go/vt/sqlparser/token.go [hunk 1]
```diff
@@ -651,7 +651,7 @@ func (tkn *Tokenizer) skipBlank() {
 func (tkn *Tokenizer) scanIdentifier(firstByte byte, isDbSystemVariable bool) (int, []byte) {
 	buffer := &bytes2.Buffer{}
 	buffer.WriteByte(firstByte)
-	for isLetter(tkn.lastChar) || isDigit(tkn.lastChar) || (isDbSystemVariable && isCarat(tkn.lastChar)) {
+	for isLetter(tkn.lastChar) || isDigit(tkn.lastChar) || tkn.lastChar == '$' || (isDbSystemVariable && isCarat(tkn.lastChar)) {
```
Comment (sougou) [inline go/vt/sqlparser/token.go]: As discussed yesterday, `$` seems to be valid at the beginning of an identifier also. So, it's better to add it to `isLetter`.

Comment (sougou) [reply go/vt/sqlparser/token.go]: Ping

**Comment by systay** [comment]
@sougou ok like this?

**Comment by sougou** [comment]
This has gone stale. I'll close this. We can have @systay rework it if the feature is still needed.

**Pull request closed without merging**